### PR TITLE
feat(misc): apply multicall3 contract code manually

### PR DESCRIPTION
**Pull request by Pangssu**
The original multicall3 contract could not be deployed to Kroma mainnet at the canonical address of 0xcA11bde05977b3631167028862bE2a173976CA11 due to an unexpected nonce increment.
This file applies the contract code to the canonical address manually in the Ecotone hardfork of Kroma.

## Summary by CodeRabbit

- **New Features**
	- Introduced a manual application feature for contract code to address deployment issues in the Ecotone hardfork, enhancing contract interaction capabilities.

- **Tests**
	- Added tests to verify the correct application of contract code under various conditions, ensuring reliability and stability.

- **Refactor**
	- Integrated new functionality into the processing and mining workflows, optimizing the application's performance and consistency.

### CODE REVIEW

File: consensus/misc/multicall3.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+// The original multicall3 contract could not be deployed to Kroma mainnet at
+// the canonical address of 0xcA11bde05977b3631167028862bE2a173976CA11 due to
+// an unexpected nonce increment.
+// This file applies the contract code to the canonical address manually in the Ecotone
+// hardfork of Kroma.
+
+var multicall3Address = common.HexToAddress("0xcA11bde05977b3631167028862bE2a173976CA11")
+
+func EnsureMulticall3(c *params.ChainConfig, timestamp uint64, db vm.StateDB) {
+	if !c.IsKroma() || c.EcotoneTime == nil || *c.EcotoneTime != timestamp {
```
Comment (0xHansLee) [inline consensus/misc/multicall3.go:20]: Why don't we make this as below?

```go
if !c.IsKroma() || c.EcotoneTime == nil || *c.EcotoneTime > timestamp || *c.EcotoneTime+2 < timestamp {
    // set code
}
```

**Review by 0xHansLee — APPROVED** [review]

**Comment by kangsorang** [comment]
Is superchain.LoadContractBytecode not available? Multicall3 is not in that repository, so I'm wondering if you injected it yourself

File: consensus/misc/multicall3.go (hunk 1 above)
Comment (0xHansLee) [reply consensus/misc/multicall3.go:20]: We are not gonna do this.

**Review by sm-stack — APPROVED** [review]

**Review by seolaoh — APPROVED** [review]
So we need to set ecotone time to odd timestamp for mainnet and even timestamp for testnet and devnet, right?

**Comment by 0xHansLee** [comment]
> So we need to set ecotone time to odd timestamp for mainnet and even timestamp for testnet and devnet, right?

Right.

**Review by kangsorang — APPROVED** [review]

**Pull request merged**